File: UnitTests/Tests/Extensions/UIColor_TRTemperatureColorsSpec.m
#import <Specta/Specta.h>
#import <Expecta/Expecta.h>
#import "UIColor+TRTemperatureColors.h"

SpecBegin(UIColor_TRTemperatureColors)

describe(@"UIColor+TRTemperatureColors", ^{
    describe(@"lighterColorByAmount:", ^{
        it(@"blends with white", ^{
            UIColor *warmer = [UIColor warmerColor];

            UIColor *blendedColor = [warmer lighterColorByAmount:0.5];

            CGFloat hue, saturation, brightness, alpha;
            [blendedColor getHue:&hue saturation:&saturation brightness:&brightness alpha:&alpha];

            expect(saturation).to.equal(0.5f);
            expect(brightness).to.beCloseToWithin(0.985, 0.001f);
        });
    });
});

SpecEnd

File: Core/Specs/Extensions/UIColor_TRTemperatureColorsSpec.swift
@testable import TroposCore
import Quick
import Nimble

final class UIColor_TRTemperatureColorsSpec: QuickSpec {
    override func spec() {
        describe("UIColor+TRTemperatureColors") {
            describe("lighterColorByAmount") {
                it("blends with white") {
                    let warmer = UIColor.warmerColor

                    let blendedColor = warmer.lighten(by: 0.5)

                    var hue: CGFloat = 0
                    var saturation: CGFloat = 0
                    var brightness: CGFloat = 0
                    var alpha: CGFloat = 0
                    blendedColor.getHue(&hue, saturation: &saturation, brightness: &brightness, alpha: &alpha)

                    expect(saturation).to(equal(0.5))
                    expect(brightness).to(beCloseTo(0.985, within: 0.001))
                }
            }
        }
    }
}
